src/hotCodeReplace.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+
+const runningSessions: Set<vscode.DebugSession> = new Set();
+
+const suppressedReasons: Set<string> = new Set();
+
+const NOT_SHOW_AGAIN: string = "Not show again";
+
+const JAVA_LANGID: string = "java";
+
+const HCR_EVENT = "hotCodeReplace";

```

Comment:
I think we need two events: hotCodeReplaceStart and hotCodeReplaceEnd. And for any event, we need the error object to see whether it's a successful one.

hotCodeReplaceStart will be used to let user know that we are in the progress of redefining classes.

Reply:
Okay, we need refine the event types for HCR.  

Currently, this event current only for error handling on the user side.

For hotCodeReplacestart/hotCodeReplaceEnd we need first add them on the debug server side.

Reply:
Refined HCR event.